lib/task_bunny/message.ex

```diff
@@ -1,9 +1,42 @@
 defmodule TaskBunny.Message do
   @moduledoc """
-  Provides functionalities to access a message and its meta data.
+  Functions to access messages and its meta data.
   """
 
   @doc """
+  Encode message body in JSON with job and arugment.
+  """
+  @spec encode(atom, any) :: String.t
+  def encode(job, payload) do
+    %{
+      "job" => Atom.to_string(job),
+      "payload" => payload,
+      "created_at" => DateTime.utc_now()
+    }
+    |> Poison.encode!
+  end
+
+  @doc """
+  Decode message body in JSON to map
+  """
+  @spec decode(String.t) :: map
+  def decode(message) do
+    case Poison.decode(message) do
+      {:ok, decoded} ->
+        job = String.to_atom(decoded["job"])
```

Comment:
Ah, yeah, I forgot to add "Elixir." prefix in original description(I changed) but that's what I intended. I prefer using "Elixir.Jobs.ExampleJob" as that's how Elixir module name should be described in atom.